### PR TITLE
Equivalent skycoin cipher tests ref #84

### LINKED ISSUE #84: Tests for hw wallet skycoin-api in compliance to Skycoin cipher API
**Feature description**

Under all circumstances `skycoin-api` must be 100% compatible with `skycoin/skycoin`

**Describe the solution you'd like**
Check for a match between `skycoin-api` input arguments and corresponding results by using values in `skycoin/skycoin` golden files.

**Additional context**
Should be limited to the subset of the Skycoin API needed in hw wallet. Maybe related to skycoin/libskycoin#7 .

**Possible implementation**

TODO: Detailed steps

### PR DESCRIPTION
**Pull request by isaacvr**
Fixes #84

 Changes:	
  - Add a [wrapper](https://en.wikipedia.org/wiki/Adapter_pattern) to skycoin-api crypto to be able to run the libskycoin tests.

 Does this change need to mentioned in CHANGELOG.md?	
 no	

 Requires testing	
 no	

 Comments about testing , should you have some

Depends on this https://github.com/skycoin/hardware-wallet/issues/259

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -34,6 +43,13 @@ before_script:
   # Install linters
   - make install-linters
   - git submodule update --remote
+  # Install gcc 6 version, required for libskycoin
+  #FIXME $TRAVIS_TAG != ''
+  - if [[ $TRAVIS_TAG == 'osx' ]]; then
+    make tiny-firmware/vendor/libskycoin/Makefile;
```
Comment (olemis) [inline .travis.yml]: why not `make -C ...` ?

Comment (olemis) [inline .travis.yml]: why not `make -C` ... ?

**Review by olemis — CHANGES_REQUESTED** [review]

File: .travis.yml [hunk 2]
```diff
@@ -11,15 +11,24 @@ matrix:
...
+  - gimme 1.12
+  - source ~/.gimme/envs/go1.12.env
   - sh "./tiny-firmware/protob/ci-scripts/install-${TRAVIS_OS_NAME}.sh" ;
   # Define env vars for Linux builds
   - if [[ $TRAVIS_OS_NAME == 'linux' ]]; then
     ls -l /usr/local/bin/protoc ;
     export CHECK_PATH="$(pwd)/check-0.12.0" ;
     export LD_LIBRARY_PATH="${LD_LIBRARY_PATH}:${CHECK_PATH}/src" ;
     export PATH="/usr/local/bin:$(pwd)/protoc/bin:$PATH" ;
+    sudo apt-get install -qq valgrind;
```
Comment (olemis) [inline .travis.yml]: why this ?

File: .travis.yml (hunk 1 above)
Comment (stdevAlDen) [reply .travis.yml]: @olemis this [solve the project](https://github.com/simelo/skywallet-mcu/blob/ad00712856cd7bab2b5963de45216979b5a6b960/Makefile#L209), then you can make use of the [`install-travis-gcc.sh`](https://github.com/fibercrypto/skywallet-mcu/pull/359/files#diff-354f30a63fb0907d4ad57269548329e3R50)

**Review by olemis — CHANGES_REQUESTED** [review]

File: .travis.yml (hunk 1 above)
Comment (olemis) [reply .travis.yml]: make a phony target instead

File: Makefile [hunk 3]
```diff
@@ -204,6 +204,27 @@ test: clean firmware tiny-firmware/bootloader/libskycoin-crypto.so ## Run all pr
 	$(MAKE) -C skycoin-api/ test && \
 	$(MAKE) VERSION_MAJOR=$(VERSION_FIRMWARE_MAJOR) VERSION_MINOR=$(VERSION_FIRMWARE_MINOR) VERSION_PATCH=$(VERSION_FIRMWARE_PATCH) GLOBAL_PATH=$(MKFILE_DIR) emulator
 	$(MAKE) EMULATOR=1 VERSION_MAJOR=$(VERSION_FIRMWARE_MAJOR) VERSION_MINOR=$(VERSION_FIRMWARE_MINOR) VERSION_PATCH=$(VERSION_FIRMWARE_PATCH) GLOBAL_PATH=$(MKFILE_DIR) -C tiny-firmware/ test
+	make test-libsky-compat
+
+tiny-firmware/vendor/libskycoin/Makefile: ## Download libskycoin for tests
```
Comment (olemis) [inline Makefile]: Change target name. Make it phony.

**Review by olemis — CHANGES_REQUESTED** [review]

File: skycoin-api/Makefile [hunk 4]
```diff
@@ -80,11 +81,18 @@ libskycoin-crypto.so: skycoin_crypto.o $(OBJS)
 libskycoin-crypto.a: skycoin_crypto.o $(OBJS)
 	$(AR) rcs libskycoin-crypto.a skycoin_crypto.o $(OBJS)
 
+libskycoin-wrapper.o: libskycoin-wrapper.c
```
Comment (olemis) [inline skycoin-api/Makefile:84]: Please explain (briefly) in README.md how running `libskycoin` test suite works and how this is used for running compatibility test suite.

**Review by olemis — CHANGES_REQUESTED** [review]

**Pull request merged**